src/main/java/ch/njol/skript/lang/SkriptParser.java

```diff
@@ -531,6 +531,9 @@ private Expression<?> parseSingleExpr(boolean allowUnparsedLiteral, @Nullable Lo
 				Expression<?> parsedExpression = parseExpression(types, expr);
 				if (parsedExpression != null) { // Expression/VariableString parsing success
 					Class<?> returnType = parsedExpression.getReturnType(); // Sometimes getReturnType does non-trivial costly operations
+					if (returnType == null)
+						throw new SkriptAPIException("Expression '" + expr + "' returned null for method Expression#getReturnType. Consider using Object.class, as null is not a valid return type.");
```

Comment:
i don't think suggesting Object.class is needed because there could be a variety of mistakes that could cause this

Reply:
Do you have an alternative suggestion?